src/game/game_init.c

```diff
@@ -609,17 +609,19 @@ void read_controller_inputs(s32 threadID) {
 
         // if we're receiving inputs, update the controller struct with the new button info.
         if (controller->controllerData != NULL) {
+            u16 changedButtons = controller->controllerData->button ^ controller->buttonDown;
```

Comment:
Make this a s32 value, as a local u16 variable here can potentially add unnecessary masking in the output code.

Reply:
Huh, really?

Also not u16?

Reply:
it probably has something to do with the fact that local non-32 bit values aren't really a thing

Reply:
C integer promotion rules mean that the result of that xor is an int, which is equivalent to an s32 value. Storing it to a u16 forces the compiler to make a decision about masking that promoted value. You can see what I mean here, where this function returns 4 instead of 2 because the result was promoted to an int.

![image](https://user-images.githubusercontent.com/68165316/174222525-0215c790-08c5-4ebe-b396-53887c6e7301.png)